runtime/runtime-core/common/test/aws/smithy/kotlin/runtime/hashing/Crc32Test.kt

```diff
@@ -20,7 +20,6 @@ class Crc32Test {
         val input = "foobar"
 
         crc.update(input.encodeToByteArray(), 0, input.length)
-        assertEquals(2666930069U, crc.digestValue()) // checksum of "foobar"
         assertEquals("nvYflQ==", crc.digest().encodeBase64String())
```

Comment:
**Question**: Why are we moving the `digestValue` checks from these tests?

Reply:
Our current JVM implementation of `digestValue()` does not reset the checksum and these tests were depending on that faulty behavior (calling `digestValue()` then `digest()` immediately after).

I've clarified the behavior in KDocs and also updated our JVM implementation. Let me know if you think we should keep the existing (arguably broken) behavior. KotlinNative / CRT will likely be unable to match that behavior though.

Reply:
Oh I see. If this was already broken behavior then you're probably right.